src/Sql/dbo/Stored Procedures/Organization_EnableCollectionEnhancements.sql

```diff
@@ -0,0 +1,299 @@
+
+                    -- Fetch the next row
+                    FETCH NEXT FROM UniqueOrgUserIdCursor INTO @Step3OrganizationUserId
+                END
+
+                CLOSE UniqueOrgUserIdCursor
+                DEALLOCATE UniqueOrgUserIdCursor;
+            COMMIT TRANSACTION;
+        END TRY
+        BEGIN CATCH
+            ROLLBACK TRANSACTION;
+            THROW;
+        END CATCH;
+
+        -- Drop the temporary table
+        DROP TABLE #TempStep3;
+
+    -- Step 4: Insert rows to [dbo].[CollectionUser] for Managers and users with 'EditAssignedCollections' permission assigned to groups with collection access
+        -- Store the results in a temporary table
+        SELECT cg.[CollectionId], ou.[Id] AS [OrganizationUserId]
+        INTO #TempStep4
+        FROM [dbo].[CollectionGroup] cg
+            INNER JOIN [dbo].[GroupUser] gu ON cg.GroupId = gu.GroupId
+            INNER JOIN [dbo].[OrganizationUser] ou ON gu.OrganizationUserId = ou.[Id]
+        WHERE (ou.[Type] = 3 OR (ou.[Permissions] IS NOT NULL AND ISJSON(ou.[Permissions]) > 0 AND JSON_VALUE(ou.[Permissions], '$.editAssignedCollections') = 'true'))
+          AND NOT EXISTS (
+            SELECT 1 FROM [dbo].[CollectionUser] cu
+            WHERE cu.[CollectionId] = cg.[CollectionId] AND cu.[OrganizationUserId] = ou.[Id]
+            );
+
+        BEGIN TRY
+            BEGIN TRANSACTION;
+                -- Insert rows into [dbo].[CollectionUser] using the temporary table
+                INSERT INTO [dbo].[CollectionUser] ([CollectionId], [OrganizationUserId], [ReadOnly], [HidePasswords], [Manage])
+                SELECT [CollectionId], [OrganizationUserId], 0, 0, 1
+                FROM #TempStep4;
+
+                -- Execute User_BumpAccountRevisionDateByOrganizationUserId for each unique OrganizationUserId
+                DECLARE @Step4OrganizationUserId UNIQUEIDENTIFIER
+
+                DECLARE UniqueOrgUserIdCursor CURSOR FOR
+                SELECT DISTINCT [OrganizationUserId]
+                FROM #TempStep4
+
+                OPEN UniqueOrgUserIdCursor
+                FETCH NEXT FROM UniqueOrgUserIdCursor INTO @Step4OrganizationUserId
+
+                WHILE (@@FETCH_STATUS = 0)
+                BEGIN
+                    -- Execute the stored procedure for the current OrganizationUserId
+                    EXEC [dbo].[User_BumpAccountRevisionDateByOrganizationUserId] @Step4OrganizationUserId
+
+                    -- Fetch the next row
+                    FETCH NEXT FROM UniqueOrgUserIdCursor INTO @Step4OrganizationUserId
+                END
+
+                CLOSE UniqueOrgUserIdCursor
+                DEALLOCATE UniqueOrgUserIdCursor;
+            COMMIT TRANSACTION;
+        END TRY
+        BEGIN CATCH
+            ROLLBACK TRANSACTION;
+            THROW;
+        END CATCH;
+
+        -- Drop the temporary table
+        DROP TABLE #TempStep4;
+
+    -- Step 5: Set all Managers to Users
```

Comment:
I think it would make more sense to do this in Steps 4 & 5 for each user affected by those steps. You could have the same UPDATE statement but joining on #TempStep4 / #TempStep5 instead. Then you save this whole step and you don't have to bump their account revision date again.

It's also how you treated clearing AccessAll.

Reply:
I have condensed steps 3, 4 and 5 into step 3.